storm-core/src/jvm/org/apache/storm/daemon/supervisor/Container.java

```diff
@@ -0,0 +1,437 @@
+            return isWindowsProcessAlive(pid, user);
+        }
+        return isPosixProcessAlive(pid, user);
+    }
+    
+    private boolean isWindowsProcessAlive(long pid, String user) throws IOException {
+        boolean ret = false;
+        ProcessBuilder pb = new ProcessBuilder("tasklist", "/nh", "/fi", "pid eq"+pid);
+        pb.redirectError(Redirect.INHERIT);
+        Process p = pb.start();
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+            if (in.readLine() != null) {
+                ret = true;
+            }
+        }
+        return ret;
+    }
+    
+    private boolean isPosixProcessAlive(long pid, String user) throws IOException {
+        boolean ret = false;
+        ProcessBuilder pb = new ProcessBuilder("ps", "-o", "user", "-p", String.valueOf(pid));
+        pb.redirectError(Redirect.INHERIT);
+        Process p = pb.start();
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+            String first = in.readLine();
+            assert("USER".equals(first));
+            String processUser;
+            while ((processUser = in.readLine()) != null) {
+                if (user.equals(processUser)) {
+                    ret = true;
+                    break;
+                } else {
+                    LOG.info("Found {} running as {}, but expected it to be {}", pid, processUser, user);
+                }
+            }
+        }
+        return ret;
+    }
+    
+    @Override
+    public boolean areAllProcessesDead() throws IOException {
+        Set<Long> pids = getAllPids();
+        String user = getWorkerUser();
+        
+        boolean allDead = true;
+        for (Long pid: pids) {
+            if (!isProcessAlive(pid, user)) {
+                LOG.warn("{}: PID {} is dead", _workerId, pid);
+            } else {
+                allDead = false;
+                break;
+            }
+        }
+        return allDead;
+    }
+
+    @Override
+    public void cleanUp() throws IOException {
+        cleanUpForRestart();
+    }
+
+    /**
+     * Setup the container to run.  By default this creates the needed directories/links in the
+     * local file system
+     * PREREQUISITE: All needed blobs and topology, jars/configs have been downloaded and
+     * placed in the appropriate locations
+     * @throws IOException on any error
+     */
+    protected void setup() throws IOException {
+        if (_port <= 0) {
+            throw new IllegalStateException("Cannot setup a container recovered with just a worker id");
+        }
+        final String topologyId = _assignment.get_topology_id();
+        if (!SupervisorUtils.doRequiredTopoFilesExist(_conf, topologyId)) {
+            LOG.info("Missing topology storm code, so can't launch  worker with assignment {} for this supervisor {} on port {} with id {}", _assignment,
+                    _supervisorId, _port, _workerId);
```

Comment:
LOG.error, and maybe just say missing required files? We don't know specifically what was missing.